modules/react/common/stories/CompoundComponentUtilities.stories.mdx

```diff
@@ -0,0 +1,301 @@
+import {Meta} from '@storybook/addon-docs';
+import {Specifications} from '@workday/canvas-kit-docs';
+import {CreateModelHook} from './examples/CreateModelHook';
+import {CreateContainer} from './examples/CreateContainer';
+import {CreateSubcomponent} from './examples/CreateSubcomponent';
+import {CreateElemPropsHook} from './examples/CreateElemPropsHook';
+import {CreateComponent} from './examples/CreateComponent';
+
+<Meta title="Hooks and Utilities/Compound Components" />
+
+# Compound Component Utilities
+
+The following utilities are used to create and compose
+[compound components](/getting-started/for-developers/resources/compound-components/).
+
+## Installation
+
+```sh
+yarn add @workday/canvas-kit-react
+```
+
+# createComponent
+
+`createComponent` is a great factory function that creates components to be exported. It enforces
+React ref forwarding, `as` prop, display name, and sub-components, and handles proper typing without
+much boiler plate. The return type is `Component<element, Props>` which looks like
+`Component<'div', Props>` which is a clean interface that tells you the default element that is
+used.
+
+## Usage
+
+`createComponent` is great to use if you want to create quick compound components or you need to set
+up components with ref forwarding.
+
+<ExampleCodeBlock code={CreateComponent} />
+
```

Comment:
I guess we also can place code block for `createComponent` function set up, before actual component. So people can copy that part without clicking on small button show code.

Reply:
This also works for `createContainer` and `createSubcomponent` functions